### DISCUSSION TITLE
Buying new things

### DISCUSSION THREAD

**Discussion by RobPattonDotCom** [inventree/InvenTree#5672] [General]
Deployed, just setting things up. I must be missing something.  Got all my base inventory imported and looks good. Now to use it!

If I purchase 100 of an item, and they are 1$ each, but the shipping for the 100 was 22$, I have been manually setting the unit price to (122/100) 1.22, though sometimes its more complicated.
100 at 1$ each + 22 shipping + 3% credit card fee.

Short of just doing all the math each time, is there an automated way to do that so inventory has the real cost?

Clearly I must be missing something, surely everyone has this issue.

Suggestions.

**Comment by SchrodingersGat** [comment]
@RobPattonDotCom it is a great question and has been asked before. But there has been no development time allocated to it. 

We have "extra line items" where you can add shipping etc but these do not get amortized across other lines. It is a complicated question, as you don't always want to necessarily split the shipping costs evenly anyway, and there may be other complications such as split shipment.

Then you have taxes which are not always handled the same way. 

So basically it is hard to work out how to implement generically.  Any suggestions?

**Reply by RobPattonDotCom** [reply]
Well, its a BIG deal for my use case.  Total cost to build a thing looks great, except that after shipping and cc fees, etc, its totally not. 

Sales tax shouldnt be an issue (for me)   but CC fees certainly are.  Its simple for me, ususally.   For the most part, Vendor 1 supplies part 1, and thats it.  Vendor 2 supplies part 2 and thats it.   So the math is fairly easy, I just feel like I shouldnt have to be doing it since I'm already putting that info into purchase orders.

The other issue with doing the math myself and just manually altering the unit cost, is that there isnt a paper trail to show any of that, and thus no reportability to show me that, for instance.   If I make widget #5, parts costs were X, and shipping those parts to me were Z.  I'm kinda afraid to know this number, but it would DEF be a valuable number to know.   If I were able to say that 8% of COGS was because of incoming shipping, It would scare me, I think.

So, 1. I'm glad to know its not somthing I've just missed, but 2. Kinda disapointed that hasnt already been dealt with, because its important.

**Reply by matmair** [reply]
I don't appreciate your tone - you are welcome to use any other software if it is this BIG of a deal for you.

**Reply by SchrodingersGat** [reply]
@RobPattonDotCom also welcome to contribute the feature yourself if it is important to you. We have a lot of features on the roadmap and only so many dev hours :)

I'd happily review a PR for this and discuss implementation details

**Reply by RobPattonDotCom** [reply]
This is the internet, there was no "tone" just relax and realize the "tone" is in your head.  I have not said one bad thing about the software, just that a feature was important to my world.

**Reply by RobPattonDotCom** [reply]
I wish it was in my skillset to help and create the feature.  I stopped coding when Turbo Pascal 6 was a thing, probly wont be of any help

**Reply by SchrodingersGat** [reply]
@RobPattonDotCom looks like there was a little bit of crossover between the age of Pascal and the age of Python - but not much!

Do you have any examples of how this approach is implemented in any other systems? I would be keen for this feature, but again is very difficult to see how it might be implemented in the general case, without being prohibitively difficult from a user-interface point of view.

For example if you purchase a milling machine and some pens, the shipping costs should not be equally split across these items - you would need a way to allocate a portion of the extra costs to each item in the order

**Comment by aortiz2323** [comment]
Hello! Has there been a more recent update to tracking shipping costs for received items or at least adding additional costs for BOM line items? Thank you!

**Reply by matmair** [reply]
Not really, the original answer from Oliver is more or less still valid. Searching in the issues will surface things like:
- https://github.com/inventree/InvenTree/issues/3928
- https://github.com/inventree/InvenTree/issues/9671
- https://github.com/inventree/InvenTree/issues/7279

IMO to get this right we need https://github.com/inventree/InvenTree/issues/8607

**Reply by matmair** [reply]
To be frank: This is a large undertaking and without either someone with proper django/python knowledge and a bunch of time or a substantial sponsoring this will not go anywhere as it won't be much fun, even for most type As
